src/serve/javaEmulators.js

```diff
@@ -58,9 +58,11 @@ function _stop(targetName) {
   return new Promise(function(resolve) {
     utils.logLabeledSuccess(emulator.name, "shutting down");
     if (emulator.instance) {
-      return emulator.instance.kill(0);
+      emulator.instance.once("exit", resolve);
+      emulator.instance.kill("SIGINT");
```

Comment:
Don't you still need to add a `resolve()` for the if case?

Reply:
I think this promise will never resolve.

Also, any reason not to send this a `SIGTERM`? I imagine they'll have the same practical effect, but semantically it may give the process an opportunity to gracefully stop.

Reply:
I'm sending the kill signal to the emulator process (`emulator.instance.kill("SIGINT")`) and then the promise resolves once the process exits (`emulator.instance.once("exit", resolve)`).

I could resolve immediately after sending the kill signal, but it's cleaner this way.

Reply:
> Also, any reason not to send this a `SIGTERM`?

[This](https://github.com/firebase/firebase-tools/blob/50215451b29732cc0efde7784b989ab68dd1c687/src/serve/javaEmulators.js#L45) expects the process to be killed by a SIGINT, which gives an exit code of 130. It makes sense, since the emulator is almost always launched from the CLI and is killed by Ctrl-c, which sends a SIGINT.

Reply:
> Probably worth thinking about what happens if the process ignores the sigint. Maybe reject the promise after some timeout?

That's a good idea. It can show a message to the user saying that the emulator process couldn't be terminated and might still be running in the background. I'll add this.

Reply:
> Maybe reject the promise after some timeout?

@ryanpbrewster Right now [this](https://github.com/firebase/firebase-tools/blob/50215451b29732cc0efde7784b989ab68dd1c687/src/serve/index.js#L33) is eating any rejection on the promise returned by `serve()`. Not sure what the best approach would be:
* Remove that altogether and allow any rejection to bubble up.
* Allow only the rejection from the timeout to bubble up.
* Leave it as is, and just `console.warn()` for that specific case.

Reply:
IIRC the reason we swallow these rejections is to avoid spilling a bunch of scary-looking error messages onto the console when it's some transient failure.

The lowest-impact change here is to just `console.warn` here. I'll defer to @mbleigh about what to do about bubbling up failures during a `serve` run.

Reply:
If you want to merge this as-is with a `console.warn()`, go ahead and re-assign this PR to me.

Reply:
I added the reject() too, besides the warning. It doesn't have any effect at the moment but I think it makes it clear that this part of the code handles some kind of failure.

Reply:
> What you'll likely want to do is `logger.debug` any error.

Good idea! Added.